soh/src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -149,11 +155,58 @@ void func_80AFB89C(EnSi* this, GlobalContext* globalCtx) {
             Item_Give(globalCtx, giveItemId);
         }
         Message_StartTextbox(globalCtx, textId, NULL);
-        Audio_PlayFanfare(NA_BGM_SMALL_ITEM_GET);
+        if (gSaveContext.n64ddFlag) {
+            Audio_PlayFanfare_Rando(getItemId);
+        } else {
+            Audio_PlayFanfare(NA_BGM_SMALL_ITEM_GET);
+        }
         this->actionFunc = func_80AFB950;
     }
 }
 
+// Function to play "get-item" fanfares according to the type of item obtained (used in rando)
+// Longer fanfares for medallions/stones/songs are behind the Cvar
+void Audio_PlayFanfare_Rando(ItemID getItemId) {
```

Comment:
seems really odd to have this living in the skull token actor, not sure the best spot for it but i feel like even `OTRGlobals` would be better

Reply:
The normal `Audio_PlayFanfare` and `Audio_PlaySoundGeneral` were already defined and used in `code_800EC960.c` so putting it there worked out without any trouble!